provisioner/puppet-masterless/provisioner.go

```diff
@@ -227,8 +275,15 @@ func (p *Provisioner) Provision(ui packer.Ui, comm packer.Communicator) error {
 	}
 
 	// Execute Puppet
+	var facterVarsString string
+	if p.config.GuestOSType == provisioner.UnixOSType {
+		facterVarsString = strings.Join(facterVars, " ")
+	} else {
+		facterVarsString = "SET " + strings.Join(facterVars, " && SET ")
+	}
+
 	p.config.ctx.Data = &ExecuteTemplate{
```

Comment:
@lfarnell if you want to submit a PR to c22:packer_4391_rework I will happily merge it, otherwise I can get to this in the next couple days (or whoever merges this from hashicorp can make those changes manually).